lightning/src/chain/keysinterface.rs

```diff
@@ -319,13 +313,15 @@ pub trait ChannelKeys : Send+Clone {
 	/// protocol.
 	fn sign_channel_announcement<T: secp256k1::Signing>(&self, msg: &UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
-	/// Set the counterparty channel basepoints and counterparty_selected/holder_selected_contest_delay.
-	/// This is done immediately on incoming channels and as soon as the channel is accepted on outgoing channels.
+	/// Set the counterparty static channel data, including basepoints,
+	/// counterparty_selected/holder_selected_contest_delay and funding outpoint.
+	/// This is done as soon as the funding outpoint is known.  Since these are static channel data,
+	/// they MUST NOT be allowed to change to different values once set.
```

Comment:
done

Reply:
The phrasing here is awkward - its not that `is_populated()` MUST be true (implying some kind of requirement on the user), but that it WILL be true (as it is a requirement for our own internal code, thus an API guarantee provided to the user).

Reply:
The `ChannelKeys` API is also used internally in an external signer.  i.e. it's called directly by user code.